src/other/tables.jl

```diff
@@ -48,6 +48,29 @@ DataFrame!(x::Vector{<:NamedTuple}) =
                         "`$(typeof(x))` without allocating new columns: use " *
                         "`DataFrame(x)` instead"))
 
+Tables.istable(::Type{<:Union{DataFrameRows,DataFrameColumns}}) = true
+Tables.columnaccess(::Type{<:Union{DataFrameRows,DataFrameColumns}}) = true
+Tables.rowaccess(::Type{<:Union{DataFrameRows,DataFrameColumns}}) = true
+Tables.columns(itr::Union{DataFrameRows,DataFrameColumns}) = Tables.columns(parent(itr))
+Tables.rows(itr::Union{DataFrameRows,DataFrameColumns}) = Tables.rows(parent(itr))
+Tables.schema(itr::Union{DataFrameRows,DataFrameColumns}) = Tables.schema(parent(itr))
+Tables.materializer(itr::DataFrameRows) =
+    eachrow ∘ prefer_singleton_callable(Tables.materializer(parent(itr)))
+function Tables.materializer(itr::DataFrameColumns)
+    f = prefer_singleton_callable(Tables.materializer(parent(itr)))
+    if eltype(itr) <: Pair
+        return x -> eachcol(f(x), true)
+    else
+        return x -> eachcol(f(x), false)
+    end
+end
+
+# A hack to workaround the type-instability of `∘`:
```

Comment:
Can you explain why this is needed?

Reply:
I illustrate this in https://github.com/JuliaLang/julia/issues/23618#issuecomment-566130606

Reply:
OK. Though a simpler solution would have been to hardcode `DataFrame` as the materializer (since that's the definition we have above).

It would be nice to at least add a comment pointing to the issue, otherwise it's quite hard to guess without context (especially if one day this gets fixed).

Reply:
> Though a simpler solution would have been to hardcode `DataFrame` as the materializer (since that's the definition we have above).

I thought somebody may have their AbstractDataFrame subtype but I guess I didn't need to worry about it.

Reply:
In theory yes, in practice it's relatively unlikely, and anyway they could just provide their custom method.

Reply:
Please see https://github.com/JuliaData/DataFrames.jl/pull/2057

Reply:
https://github.com/JuliaData/DataFrames.jl/pull/2057